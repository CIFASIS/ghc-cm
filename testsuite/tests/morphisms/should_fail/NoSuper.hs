module Main where

class A a where a :: a -> Int
class C c where c :: c -> Int
class C b => B b where b :: b -> Int

-- m :: AD a -> [[CD a]] -> BD a
-- should fail here, no way to generically get C
class morphism A -> B where
    b x = 2222

-- Explodes because no instance for C Int (or similar)
instance A Int where
    a x = x

main = print $ b (2 :: Int)
