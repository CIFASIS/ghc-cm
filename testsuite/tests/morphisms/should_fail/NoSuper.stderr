
NoSuper.hs:9:1: error:
    • Bad morphism declaration
    • No generic way to obtain an instance for the C b superclass of B b
      In the morphism declaration
        ‘class morphism A -> B -- Defined at NoSuper.hs:9:1’
